### ISSUE TITLE
Dotnet restore fails behind corporate proxy

### ISSUE THREAD

**Issue by kkokosa** [dotnet/cli#2190]
## Steps to reproduce

Create new Dockerfile as follows:

```
FROM microsoft/dotnet

ENV http_proxy http://user:login@host:port
ENV https_proxy http://user:login@host:port

COPY ./Proxy_CA.crt /usr/local/share/ca-certificates/Proxy_CA.crt
RUN ["sudo", "update-ca-certificates"]
CMD ["bash"]
```

Then build new image:

`sudo docker build -t dotnet-cmd .`
## Actual behavior

Run a new container:

`sudo docker run -it dotnet-cmd`

Then, inside container:

```
root@b268392801e3:/# mkdir temp
root@b268392801e3:/# cd temp
root@b268392801e3:/temp# dotnet new
Created new C# project in /temp.
root@b268392801e3:/temp# dotnet restore
log  : Restoring packages for /temp/project.json...
error: The reason phrase must not contain new-line characters.
```

while normal curl works:

```
root@b268392801e3:/temp# curl https://dotnet.myget.org/F/dotnet-core/api/v3/index.json
{"version":"3.0.0-beta.1", ...
root@b268392801e3:/temp# curl https://api.nuget.org/v3/index.json                     
{
  "version": "3.0.0-beta.1", ...
```

Full verbose output from dotnet restore:

```
root@b268392801e3:/temp# dotnet restore -v debug
trace: Running restore with 2 concurrent jobs.
trace: Reading project file /temp/project.json.
trace: Loaded project temp from /temp/project.json.
trace: Found project root directory: /temp.
trace: Using packages directory: /root/.nuget/packages/.
log  : Restoring packages for /temp/project.json...
trace: Restoring packages for DNXCore,Version=v5.0...
error: The reason phrase must not contain new-line characters.
trace: System.AggregateException: One or more errors occurred. (The reason phrase must not contain new-line characters.) ---> System.FormatException: The reason phrase must not contain new-line characters.
trace:    at System.Net.Http.HttpResponseMessage.set_ReasonPhrase(String value)
trace:    at System.Net.Http.CurlResponseParseUtils.ReadStatusLine(HttpResponseMessage response, String responseHeader)
trace:    at System.Net.Http.CurlHandler.TryParseStatusLine(HttpResponseMessage response, String responseHeader, EasyRequest state)
trace:    at System.Net.Http.CurlHandler.MultiAgent.CurlReceiveHeadersCallback(IntPtr buffer, UInt64 size, UInt64 nitems, IntPtr context)
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.HttpRetryHandler.<SendAsync>d__13.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.HttpSource.<SendWithCredentialSupportAsync>d__25.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.HttpSource.<GetThrottled>d__20.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.HttpSource.<GetAsync>d__18.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.v3.ServiceIndexResourceV3Provider.<GetIndexJson>d__8.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.v3.ServiceIndexResourceV3Provider.<TryCreate>d__9.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.Types.SourceRepository.<GetResourceAsync>d__11`1.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.Types.SourceRepository.<GetResourceAsync>d__10`1.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.v3.RemoteRepositories.HttpFileSystemBasedFindPackageByIdResourceProvider.<TryCreate>d__2.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.Types.SourceRepository.<GetResourceAsync>d__11`1.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.Types.SourceRepository.<GetResourceAsync>d__10`1.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.SourceRepositoryDependencyProvider.<EnsureResource>d__13.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.SourceRepositoryDependencyProvider.<FindLibraryAsync>d__8.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<>c__DisplayClass15_1.<<FindLibrary>b__0>d.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<FindLibrary>d__15.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<FindLibraryByVersion>d__14.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<FindLibraryMatch>d__11.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<FindLibraryEntry>d__8.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<CreateGraphNode>d__3.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<CreateGraphNode>d__3.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Commands.RestoreCommand.<WalkDependenciesAsync>d__18.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Commands.RestoreCommand.<TryRestore>d__9.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Commands.RestoreCommand.<ExecuteRestoreAsync>d__8.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Commands.RestoreCommand.<ExecuteAsync>d__7.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.CommandLine.XPlat.Program.<ExecuteRestoreAsync>d__14.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.CommandLine.XPlat.Program.<>c__DisplayClass7_2.<<Main>b__4>d.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.CommandLine.XPlat.Program.<CompleteTaskAsync>d__10.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.CommandLine.XPlat.Program.<>c__DisplayClass7_1.<<Main>b__3>d.MoveNext()
trace:    --- End of inner exception stack trace ---
trace:    at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
trace:    at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
trace:    at Microsoft.Dnx.Runtime.Common.CommandLine.CommandLineApplication.Execute(String[] args)
trace:    at NuGet.CommandLine.XPlat.Program.Main(String[] args)
trace: ---> (Inner Exception #0) System.FormatException: The reason phrase must not contain new-line characters.
trace:    at System.Net.Http.HttpResponseMessage.set_ReasonPhrase(String value)
trace:    at System.Net.Http.CurlResponseParseUtils.ReadStatusLine(HttpResponseMessage response, String responseHeader)
trace:    at System.Net.Http.CurlHandler.TryParseStatusLine(HttpResponseMessage response, String responseHeader, EasyRequest state)
trace:    at System.Net.Http.CurlHandler.MultiAgent.CurlReceiveHeadersCallback(IntPtr buffer, UInt64 size, UInt64 nitems, IntPtr context)
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.HttpRetryHandler.<SendAsync>d__13.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.HttpSource.<SendWithCredentialSupportAsync>d__25.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.HttpSource.<GetThrottled>d__20.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.HttpSource.<GetAsync>d__18.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.v3.ServiceIndexResourceV3Provider.<GetIndexJson>d__8.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.v3.ServiceIndexResourceV3Provider.<TryCreate>d__9.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.Types.SourceRepository.<GetResourceAsync>d__11`1.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.Types.SourceRepository.<GetResourceAsync>d__10`1.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.v3.RemoteRepositories.HttpFileSystemBasedFindPackageByIdResourceProvider.<TryCreate>d__2.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.Types.SourceRepository.<GetResourceAsync>d__11`1.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Protocol.Core.Types.SourceRepository.<GetResourceAsync>d__10`1.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.SourceRepositoryDependencyProvider.<EnsureResource>d__13.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.SourceRepositoryDependencyProvider.<FindLibraryAsync>d__8.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<>c__DisplayClass15_1.<<FindLibrary>b__0>d.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<FindLibrary>d__15.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<FindLibraryByVersion>d__14.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<FindLibraryMatch>d__11.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<FindLibraryEntry>d__8.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<CreateGraphNode>d__3.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.DependencyResolver.RemoteDependencyWalker.<CreateGraphNode>d__3.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Commands.RestoreCommand.<WalkDependenciesAsync>d__18.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Commands.RestoreCommand.<TryRestore>d__9.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Commands.RestoreCommand.<ExecuteRestoreAsync>d__8.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.Commands.RestoreCommand.<ExecuteAsync>d__7.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.CommandLine.XPlat.Program.<ExecuteRestoreAsync>d__14.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.CommandLine.XPlat.Program.<>c__DisplayClass7_2.<<Main>b__4>d.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.CommandLine.XPlat.Program.<CompleteTaskAsync>d__10.MoveNext()
trace: --- End of stack trace from previous location where exception was thrown ---
trace:    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
trace:    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
trace:    at NuGet.CommandLine.XPlat.Program.<>c__DisplayClass7_1.<<Main>b__3>d.MoveNext()<---

```

**Note:** without configuring Proxy_CA.crt `curl` is failing with the following error so I believe `dotnet` is somehow ignoring it, while `curl` does not:

```
root@4c12f5a8da73:/test# curl https://dotnet.myget.org/F/dotnet-core/api/v3/index.json
curl: (60) SSL certificate problem: self signed certificate in certificate chain
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
```

**Note:** our proxy is using Active Directory credentianls, provided by the `user:login` in the proxy settings.
## Environment data

`dotnet --info` output:

```
root@b268392801e3:/temp# dotnet --version
.NET Command Line Tools (1.0.0-beta-001598)

Product Information:
 Version:     1.0.0-beta-001598
 Commit Sha:  N/A

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  14.04
 OS Platform: Linux
 Runtime Id:  ubuntu.14.04-x64
```

**Comment by carlosrfernandez** [comment]
Same here... Right now it's a blocking issue for us. I hope it's fixed soon.

**Comment by danpantry** [comment]
:+1: this is extremely annoying. The worst part is that I have developed some code at home that works except for some services not working because I'm not on the corporate network, but as soon as I'm on the corporate network I can't run the project because the network is blocking NuGet unless it's through the proxy.. which `dotnet restore` is ignoring!

If there's anything I can do to help contribute + push this along I will because right now it is locking the entire of the "new" style of .NET from us

**Comment by harikm86** [comment]
Can you try setting the Proxy settings in nuget.config? More information available here https://docs.nuget.org/consume/nuget-config-settings#proxy-settings

**Comment by danpantry** [comment]
I did try that, unfortunately, though that may be more of an issue with our set up than the cli.

**Comment by DavidADoran** [comment]
I'm seeing a similar thing when using DNU on Ubuntu. The corporate proxy normally expects Active Directory credentials, so in order to get anything to work, curl, dnu etc., I've had the guys grant my Ubuntu VM exemption from the credential check.

That made things work OK until I updated to 1.0.0-rc1-update1, where it now fails. I have both versions of DNX on there:
```
dnvm list

Active Version              Runtime Architecture OperatingSystem Alias
------ -------              ------- ------------ --------------- -----
       1.0.0-beta7          coreclr x64          linux
  *    1.0.0-rc1-update1    coreclr x64          linux
```  
When using beta7, dnu restore works OK. E.g. below (I've redacted stuff I shouldn't be showing here!)
```
dnu restore
Microsoft .NET Development Utility CoreCLR-x64-1.0.0-beta7-15532
Restoring packages for...
Writing lock file...
Restore complete, 675ms elapsed
```

However, when trying the same with rc1-update1 I get lots of this:
```
dnu restore
Microsoft .NET Development Utility CoreClr-x64-1.0.0-rc1-16231
  GET https://api.nuget.org/v3-flatcontainer/system.linq/index.json
Warning: FindPackagesById: System.Linq
  The value cannot be null or empty.
Parameter name: UserName
```

The bit at the end, "The value cannot be null or empty.\n Parameter name: UserName" is what makes me thing this is the proxy again. Also I'm pretty sure this worked fine from my home PC, unfettered by a proxy.

The reported issue is for the .NET CLI tools but I understand they're just layers over the old dnx and dnu code.  From the above it seems that the issue I'm seeing, probably related if not the same, appeared somewhere between beta7 and rc1-update1. Hopefully that's a useful clue.

**Comment by vip32** [comment]
after setting the Proxy settings in nuget the issue is resolved.

**Comment by blackdwarf** [comment]
@kkokosa is this still reproing? If yes, this issue should be moved to NuGet/Home as `dotnet restore` uses Nuget to restore packages.

**Comment by tebeco** [comment]
@blackdwarf 
Who else should I tag ? Nuget seems to works properly without specific settings

# Versions #
>### nuget
NuGet Version: 3.5.0.1737
>### nuget update -self
Checking for updates from https://www.nuget.org/api/v2/.
Currently running NuGet.exe 3.5.0-rtm-1737.
NuGet.exe is up to date.

>### dotnet --version
1.0.0-preview2-003131

# Project #
Working with .Net Core App
Add a nuget that you don't have in Visual Studio Cache (I've used NUnit 3.5) in project.json

# Restore #
## Dotnet CLI ##
>### dotnet restore
log  : Restoring packages for D:\Sources\Other\NetCoreConsoleapp\src\NetCoreConsoleapp\project.json...
error: Unable to load the service index for source https://api.nuget.org/v3/index.json.
error:   Response status code does not indicate success: 407 (Proxy Authentication Required).

## Nuget CLI ##
TL;DR ==> Working
>### nuget restore
MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'.
Restoring packages for D:\Sources\Other\NetCoreConsoleapp\src\NetCoreConsoleapp\project.json...
...

**Comment by piotrpMSFT** [comment]
Looks like this got fixed. If anyone is still experiencing this issue please reopen with details at https://github.com/nuget/home
